Results page and PXP support for Hepatitis C

mpbrown opened the pull request:
# FRONTEND PULL REQUEST

## Related Issue

- Resolves #7437 and #7441 

## Changes Proposed

- Adds translation constants for showing Hepatitis-C result on Print Details and PXP View Result
- Both English and Spanish translations
- Adds gender of sexual partners response on Hepatitis-C View Details modal

## Additional Information

- Result guidance is shown for **any** Hep C result, not just positives
- Review and merge **after** #8226 

## Testing

- Will be deployed on dev4

## Screenshots / Demos

- Hepatitis-C already available as a selectable condition on Results page
![Screenshot 2024-10-29 105549](https://github.com/user-attachments/assets/e15dc214-f1d2-4ca2-a8dc-3a9270b6e7f5)

- Hepatitis-C already available as a condition in the Download Results spreadsheet
![Screenshot 2024-10-29 105620](https://github.com/user-attachments/assets/b53bca5e-03ef-4de4-ad5d-cb4f56f4cb12)

- Hepatitis-C on Print Details modal
![Screenshot 2024-10-29 105255](https://github.com/user-attachments/assets/5624d42d-11ff-460c-8bf5-20f68715ce03)

- Spanish translations for Hepatitis-C on Print Details modal
![Screenshot 2024-10-29 105302](https://github.com/user-attachments/assets/94fbb05f-c60e-476a-9e96-586e21f6e117)

- Hepatitis-C and AOE responses on View Details modal
![Screenshot 2024-10-29 105448](https://github.com/user-attachments/assets/559428a9-fe82-42dc-a884-75db6dcd6d99)

- Hepatitis-C result on the PXP View Result
![Screenshot 2024-10-29 105232](https://github.com/user-attachments/assets/b3c61525-cdfa-4829-9300-9cbb44f76281)

- Spanish translations for Hepatitis-C result on the PXP View Result
![Screenshot 2024-10-29 105237](https://github.com/user-attachments/assets/75c48fe9-7046-49e5-8a32-eed3f43cc67b)

- Negative Hepatitis-C result with result guidance still shown
![image](https://github.com/user-attachments/assets/4d8a34c5-b29f-4e52-a0f5-d43775b67978)

mehansen commented on frontend/src/lang/es.ts:
```diff
@@ -43,6 +43,7 @@ export const es: LanguageConfig = {
         RSV: "VRS",
         HIV: "VIH",
         SYPHILIS: "Sífilis",
+        HEPATITIS_C: "Hepatitis C",
```
should there be a dash in this string so it matches the name in other places?

mpbrown replied:
Checked with Jayna on this and we should actually switch everything else to use no dash so that it matches how the CDC refers to it as "Hepatitis C"

mpbrown replied:
@mehansen any preference for whether I make those naming changes in this current PR? I created a db changeset but debating whether to have it be a separate PR for simpler code review

mpbrown replied:
[Follow up PR created here](https://github.com/CDCgov/prime-simplereport/pull/8255)

kenieh commented:
LGTM!

emyl3 reviewed (dismissed):
hahah I thought I was losing my mind seeing all the same changes that I thought I already reviewed in the previous PR. 😵  Didn't realize this was not up to date with `main` 😅 so  reviewed the diff locally and codewise looks good!

Testing on dev4 also worked as expected! Nice work on this, Mike! 🎉

bobbywells52 reviewed (approved):
Tested on dev 4 and looks great -- thanks Mike!

mehansen reviewed (approved):
changes look good but can you please rebase from main next time before opening a PR? this PR is really hard to read
also curious if git is smart enough to not apply the same changes again in the second merge

mpbrown commented:
@emyl3 @mehansen sorry I totally forgot to rebase this one on main after the other hep C PR got merged! 😬 I added those other quick fixes and rebased so it's ready for re-review

emyl3 reviewed (approved):
Looks good! Thanks for addressing Merethe's nits!
